lib/base/heimbase-atomics.h

```diff
@@ -164,6 +181,10 @@ heim_base_exchange_64(uint64_t *p, uint64_t newval)
 #define heim_base_exchange_32(t,v)	((ULONG)InterlockedExchange((LONG volatile *)(t), (LONG)(v)))
 #define heim_base_exchange_64(t,v)	((ULONG64)InterlockedExchange64((LONG64 violatile *)(t), (LONG64)(v)))
 
+#define heim_base_cas_pointer(t,e,d) 	InterlockedCompareExchangePointer((atomic_l)(t), (d), (e))
+#define heim_base_cas_32(t,e,d)		InterlockedCompareExchange  ((atomic_l)(t), (d), (e))
+#define heim_base_cas_64(t,e,d)		InterlockedCompareExchange64((atomic_l)(t), (d), (e))
+
```

Comment:
Windows will want `PVOID volatile *`, `LONG volatile *`, `LONG64 volatile *` respectively. If they even need casts?

Reply:
Oops, good catch!

A test that uses these (for what?) would expose this.